src/components/Choropleth.vue

```diff
@@ -150,59 +94,54 @@ const tooltipForFeature = (feature: Feature) => {
-                // select feature's country, or unselect if click on it when already selected
-                let countryToSelect: string;
-                if (country === selectedCountryId.value) {
-                    countryToSelect = "";
-                } else {
-                    countryToSelect = country;
-                }
-                router.push(`/${APP_BASE_ROUTE}/${selectedIndicator.value}/${countryToSelect}`);
-            }
-        });
-    }
+const createTooltips = (feature: Feature, layer: Layer) => {
+    layer.bindTooltip(tooltipForFeature(feature));
+    layer.on({
+        click: async () => {
+            waitingForMapBounds.value = true;
+            const country = feature.properties[featureProperties.country];
+            // select feature's country, or unselect if click on it when already selected
+            const countryToSelect = country === selectedCountryId.value ? "" : country;
+            debounce(() => router.push(`/${APP_BASE_ROUTE}/${selectedIndicator.value}/${countryToSelect}`))();
+        }
+    });
 };
 
-const borderColor = (fillColor: string) => {
-    // for drawing borders more subtly, desaturate and fade the
-    // color returned from color scale in rgb format
+const fadeColour = (fillColor: string, desaturate = 0.7, fade = 0.7) => {
+    // for drawing borders more subtly and fading our features, desaturate
+    // and fade the color returned from color scale in rgb format
     const c = Color(fillColor);
-    return c.desaturate(0.7).fade(0.7).rgb();
+    return c.desaturate(desaturate).fade(fade).rgb();
 };
 
-const updateTooltips = () => {
-    featureRefs.value.forEach((geojson) => {
-        if (geojson.geojson && geojson.leafletObject) {
-            const f: FeatureWithColour = featuresWithColours.value[getFeatureId(geojson.geojson)];
-            if (f && f.feature) {
-                geojson.leafletObject.eachLayer((layer: Layer) => {
-                    layer.setTooltipContent(tooltipForFeature(f.feature));
-                });
-            }
-        }
-    });
+const style = (f: Feature) => {
+    const isFaded = selectedCountryId.value && f.properties![featureProperties.country] !== selectedCountryId.value;
+    const featureColour = getColourForFeature(f, selectedIndicator.value);
+    const fillColor = isFaded ? fadeColour(featureColour, 0.5, 0.6) : featureColour;
+    return { className: "geojson", fillColor, color: fadeColour(fillColor) };
 };
 
-const updateMap = () => {
-    updateTooltips();
-};
+const updateMap = async (newFeatures: Feature[]) => {
+    if (!map.value) return;
+
+    // remove layer from map
+    geoJsonLayer.value.remove();
+
+    // create new geojson and add to map
+    geoJsonLayer.value = geoJSON<FeatureProperties, Geometry>(newFeatures, {
+        style,
+        onEachFeature: createTooltips
+    }).addTo(map.value.leafletObject);
+
+    // update bounds
+    const country = selectedCountryId.value || "GLOBAL";
```

Comment:
Ha, I just suggested to @david-mears-2 that it might be useful to call updateBounds in implementing the map reset button... so it might be nice to keep that as its own method.,

Reply:
sure can just pull it out into a function, i am assuming this will need to be further extracted into a composable as part of the map reset PR for david to be able to use this in another component mainly because `bounds` is not in the store? or maybe we should put `bounds` in the store and david can just use the updateBounds function?

Reply:
We were just going to make the reset button emit an event and let the choro handle it with its own `updateBounds`